### DISCUSSION TITLE
Upgrad to hyper v1

### DISCUSSION THREAD

**Discussion by CriusDe** [hyperium/hyper#3564] [Q&A]
Hello,

I am currently trying to upgrade from hyper 0.14x to v1 and encountered some errors I wasn't able to resolve on my own. 

First I use the legacy client and am wondering what the return type is i should be using (is it BoxBody or does it depend on what I need?)

```
pub fn hyper_client(insecure: bool) -> Client<HttpsConnector<HttpConnector>,  **BoxBody<Bytes, Error>**> {
    let mut tls = rustls::ClientConfig::builder()
        .with_safe_defaults()
        .with_native_roots()
        .with_no_client_auth();

    if insecure {
        let mut dangerous = tls.dangerous();
        dangerous.set_certificate_verifier(SkipServerVerification::new());
    }

    let https = hyper_rustls::HttpsConnectorBuilder::new()
        .with_tls_config(tls)
        .https_or_http()
        .enable_http1()
        .build();

    Client::builder(TokioExecutor::new()).build::<_,  **BoxBody<Bytes, Error>**>(https)
}
```
Then when I want to create a request e.g: 
```
let mut req = hyper::Request::builder()
        .method(Method::POST)
        .header("Authorization", credentials.encode())
        .header("content-type", "application/json")
        .uri(uri)
        .body(Body::from(serde_json::to_string(&param.0).unwrap()))
        .unwrap();

    // global::get_text_map_propagator(|propagator| {
    //     propagator.inject_context(&cx, &mut HeaderInjector(req.headers_mut()))
    // });

    Ok(state.proxy_api.client.request(**req**).await?)
```
I get the error **expected `Request<BoxBody<Bytes, Error>>`, found `Request<Body>`** which I can understand, but the Body type is not longer available in hyper v1. Axum has its own body type but I don't think I can use this here. 

------------------------

Another thing im still trying to figure out, is how I can get from `Response<Incoming>` a byte representation like before: 
`
let bytes = hyper::body::to_bytes(resp.into_body()).await?`

I could not find anything ragarding this problem.

**Comment by seanmonstar** [comment]
Have you had a look at the upgrading guide? https://hyper.rs/guides/1/upgrading/

If something in there is unclear, we can try to make it better.
